### PR TITLE
Feature/xaptc 63

### PR DESCRIPTION
**Pull request by CameliaFarcas**

### CODE REVIEW

File: uiServices/src/homepage_test.go [hunk 1]
```diff
@@ -1,61 +1,53 @@
 package services
 
-import (
-	"github.com/go-chi/chi"
-	"github.com/stretchr/testify/assert"
-	"net/http"
-	"net/http/httptest"
-	"testing"
-)
+// func TestStartUiModeSuccesfully(t *testing.T) {
```
Comment (ConorSmyth) [inline uiServices/src/homepage_test.go]: @jenny-flynn These tests rely on the ui distribution folder being generated along with index.html. I don't think they are adding much value maybe , could we remove them. Unless you have ideas on how to save them .

Comment (ConorSmyth) [reply uiServices/src/homepage_test.go]: I was also the one that asked @CameliaFarcas to comment them out so that we could get your opinion

File: uiServices/src/configServices.go [hunk 2]
```diff
@@ -193,3 +195,19 @@ func putConfigs(rw http.ResponseWriter, req *http.Request) {
...
+	configFileName := chi.URLParam(req, "configFileName")
+	newConfigFileName := chi.URLParam(req, "newConfigFileName")
+
+	if len(newConfigFileName) < 1 {
+		logrus.Error("File path does not exist")
+		rw.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	os.Rename(fmt.Sprintf("%s%s.xml", path, configFileName), fmt.Sprintf("%s%s.xml", path, newConfigFileName))
```
Comment (jenny-flynn) [inline uiServices/src/configServices.go]: made comments about os.Rename in another PR.

File: uiServices/src/homepage.go [hunk 3]
```diff
@@ -104,6 +105,7 @@ func routeConfigs() http.Handler {
 	router.Post("/", postConfigs)
 	router.Get("/{configName}", getConfigs)
 	router.Put("/{configName}", putConfigs)
+	router.Put("/filename/{configFileName}/{newConfigFileName}", putConfigFileName)
```
Comment (jenny-flynn) [inline uiServices/src/homepage.go:117]: not sure i would have gone with this approach for renaming the file i.e. creating a new route

File: uiServices/src/homepage.go [hunk 4]
```diff
@@ -128,6 +130,7 @@ func routeTestCases() http.Handler {
 	router.Get("/", getAllTestCases)
 	router.Get("/{testCaseName}", getTestCase)
 	router.Delete("/{testCaseName}", deleteTestCase)
+	router.Delete("/all", deleteAllTestCases)
```
Comment (jenny-flynn) [inline uiServices/src/homepage.go]: this would be better as router.Delete("/", deleteAllTestCases)

File: uiServices/src/homepage_test.go (hunk 1 above)
Comment (jenny-flynn) [reply uiServices/src/homepage_test.go]: yep, i've no issues with this being removed. it's tested in the godog tests anyway

File: uiServices/src/configServices.go (hunk 2 above)
Comment (ConorSmyth) [reply uiServices/src/configServices.go]: @jenny-flynn I can't find your other comment. 


This is the code now 

```
	configFilePath := fmt.Sprintf("%s%s.xml", path, configFileName)
	newConfigFilePath := fmt.Sprintf("%s%s.xml", path, newConfigFileName)

	err := os.Rename(configFilePath, newConfigFilePath)
	if err != nil {
		logrus.Error("File was not updated", err)
		rw.WriteHeader(http.StatusInternalServerError)
		return
	}
```

File: uiServices/src/homepage_test.go (hunk 1 above)
Comment (jenny-flynn) [reply uiServices/src/homepage_test.go]: @CameliaFarcas can you remove these?

File: uiServices/src/testCaseServices.go [hunk 5]
```diff
@@ -292,3 +292,29 @@ func deleteTestCase(rw http.ResponseWriter, req *http.Request) {
...
+		filepath := fmt.Sprintf("%s%s", testCasePathDir, file.Name())
+
+		err := os.Remove(filepath)
+		if err != nil {
+			logrus.Errorf("Error deleting the files from directory: %s", err)
+			rw.WriteHeader(http.StatusNotFound)
+			return
+		}
+
+		rw.WriteHeader(http.StatusOK)
```
Comment (jenny-flynn) [inline uiServices/src/testCaseServices.go]: move this outside for loop

Comment (jenny-flynn) [inline uiServices/src/testCaseServices.go]: http.StatusNoContent would be better

File: uiServices/src/testCaseServices.go [hunk 6]
```diff
@@ -292,3 +292,29 @@ func deleteTestCase(rw http.ResponseWriter, req *http.Request) {
 	rw.WriteHeader(http.StatusNoContent)
 
 }
+func deleteAllTestCases(rw http.ResponseWriter, req *http.Request) {
+	testCasePathDir := getTestCaseHeader(req)
+	logrus.Println("testCasePathDir", testCasePathDir)
+	if err := IsHeaderValid(testCasePathDir); err != nil {
+		rw.WriteHeader(http.StatusBadRequest)
```
Comment (AlexeiSuleac) [inline uiServices/src/testCaseServices.go]: Probably a good idea to log the error here

File: uiServices/src/testCaseServices.go [hunk 7]
```diff
@@ -292,3 +292,29 @@ func deleteTestCase(rw http.ResponseWriter, req *http.Request) {
...
+		return
+	}
+
+	files, err := ioutil.ReadDir(testCasePathDir)
+	if err != nil {
+		logrus.Error(err)
```
Comment (AlexeiSuleac) [inline uiServices/src/testCaseServices.go]: Should you not try and handle this error in some way?

File: uiServices/src/homepage_test.go [hunk 8]
```diff
@@ -1,61 +0,0 @@
-package services
```
Comment (jenny-flynn) [inline uiServices/src/homepage_test.go]: why is this file deleted?

File: uiServices/src/testCaseServices.go [hunk 9]
```diff
@@ -292,3 +292,28 @@ func deleteTestCase(rw http.ResponseWriter, req *http.Request) {
...
+		filepath := fmt.Sprintf("%s%s", testCasePathDir, file.Name())
+
+		err := os.Remove(filepath)
+		if err != nil {
+			logrus.Errorf("Error deleting the files from directory: %s", err)
+			rw.WriteHeader(http.StatusNotFound)
+			return
+		}
+	}
+	rw.WriteHeader(http.StatusOK)
```
Comment (jenny-flynn) [inline uiServices/src/testCaseServices.go]: http.StatusNoContent

**Comment by CameliaFarcas** [comment]
@jenny-flynn  I have deleted the homepage_test.go file as all code was commented out, so I thought there is no reason to keep a file with no unit tests in it

File: uiServices/src/testCaseServices.go [hunk 10]
```diff
@@ -294,24 +307,27 @@ func deleteTestCase(rw http.ResponseWriter, req *http.Request) {
...
 	for _, file := range files {
 		filepath := fmt.Sprintf("%s%s", testCasePathDir, file.Name())
 
 		err := os.Remove(filepath)
 		if err != nil {
-			logrus.Errorf("Error deleting the files from directory: %s", err)
-			rw.WriteHeader(http.StatusNotFound)
+			logrus.Error("Error removing the files from directory", err)
+			rw.WriteHeader(http.StatusNoContent)
 			return
```
Comment (jenny-flynn) [inline uiServices/src/testCaseServices.go]: failure to remove a file should fail silently or return an internalserver error.

File: uiServices/src/testCaseServices.go [hunk 11]
```diff
@@ -294,24 +307,27 @@ func deleteTestCase(rw http.ResponseWriter, req *http.Request) {
...
 		}
 	}
```
Comment (jenny-flynn) [inline uiServices/src/testCaseServices.go]: move `rw.WriteHeader(http.StatusNoContent)` here

File: uiServices/src/testCaseServices.go [hunk 12]
```diff
@@ -294,24 +307,27 @@ func deleteTestCase(rw http.ResponseWriter, req *http.Request) {
...
 	files, err := ioutil.ReadDir(testCasePathDir)
 	if err != nil {
-		logrus.Error(err)
+		logrus.Error("Error reading the directory ", err)
+		rw.WriteHeader(http.StatusInternalServerError)
+		return
 	}
 
 	for _, file := range files {
 		filepath := fmt.Sprintf("%s%s", testCasePathDir, file.Name())
```
Comment (jenny-flynn) [inline uiServices/src/testCaseServices.go]: first, we should verify that the files are test cases -- see GET ALL testcases for example

this can be a separate card though :)

File: uiServices/src/testCaseServices.go (hunk 10 above)
Comment (CameliaFarcas) [reply uiServices/src/testCaseServices.go]: @jenny-flynn   you suggested yesterday that http.StatusNoContent would be better

Comment (jenny-flynn) [reply uiServices/src/testCaseServices.go]: yep sorry i misread the err != nil condition. internalservererror here, and no content if err == nil e.g. the last line of the function. sorry!

File: uiServices/src/testCaseServices.go (hunk 12 above)
Comment (CameliaFarcas) [reply uiServices/src/testCaseServices.go]:
```
if testCase.TestName != "" {
				testCases = append(testCases, Case{
					Name:        testCase.TestName,
					Description: testCase.Description,
					HttpMethod:  testCase.HTTPMethod,
				})
			}
```
is this what I should follow?
